pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -136,10 +136,13 @@ func (r *ReplicaSetReconciler) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	r.log.Debug("Building service")
-	svc := buildService(mdb)
-	if err = r.client.CreateOrUpdate(&svc); err != nil {
-		r.log.Infof("The service already exists... moving forward: %s", err)
+	r.log.Debug("Building and creating service")
+	if err = r.createService(mdb); err != nil {
```

Comment:
as mentioned below, we may not need this `createService` function at all, it should be safe to use `r.client.Create` directly